### ISSUE TITLE
Commit fdf4c3e90 breaks loading existing RoBERTa pretrained models

### ISSUE THREAD

**Issue by sarahwie** [pytorch/fairseq#1320]
Due to logic in `q_proj`,`k_proj`,`v_proj` weights being changed from `in_proj`.

Full traceback:
```
roberta = RobertaModel.from_pretrained('./roberta.large', checkpoint_file='model.pt')

>> RuntimeError: Error(s) in loading state_dict for RobertaModel:
	Missing key(s) in state_dict: "decoder.sentence_encoder.layers.0.self_attn.k_proj.weight", "decoder.sentence_encoder.layers.0.self_attn.k_proj.bias", "decoder.sentence_encoder.layers.0.self_attn.v_proj.weight", "decoder.sentence_encoder.layers.0.self_attn.v_proj.bias", "decoder.sentence_encoder.layers.0.self_attn.q_proj.weight", "decoder.sentence_encoder.layers.0.self_attn.q_proj.bias", "decoder.sentence_encoder.layers.1.self_attn.k_proj.weight", "decoder.sentence_encoder.layers.1.self_attn.k_proj.bias", "decoder.sentence_encoder.layers.1.self_attn.v_proj.weight", "decoder.sentence_encoder.layers.1.self_attn.v_proj.bias", "decoder.sentence_encoder.layers.1.self_attn.q_proj.weight", "decoder.sentence_encoder.layers.1.self_attn.q_proj.bias", "decoder.sentence_encoder.layers.2.self_attn.k_proj.weight", "decoder.sentence_encoder.layers.2.self_attn.k_proj.bias", "decoder.sentence_encoder.layers.2.self_attn.v_proj.weight", "decoder.sentence_encoder.layers.2.self_attn.v_proj.bias", "decoder.sentence_encoder.layers.2.self_attn.q_proj.weight", "decoder.sentence_encoder.layers.2.self_attn.q_proj.bias", "decoder.sentence_encoder.layers.3.self_attn.k_proj.weight", "decoder.sentence_encoder.layers.3.self_attn.k_proj.bias", "decoder.sentence_encoder.layers.3.self_attn.v_proj.weight", "decoder.sentence_encoder.layers.3.self_attn.v_proj.bias", "decoder.sentence_encoder.layers.3.self_attn.q_proj.weight", "decoder.sentence_encoder.layers.3.self_attn.q_proj.bias", "decoder.sentence_encoder.layers.4.self_attn.k_proj.weight", "decoder.sentence_encoder.layers.4.self_attn.k_proj.bias", "decoder.sentence_encoder.layers.4.self_attn.v_proj.weight", "decoder.sentence_encoder.layers.4.self_attn.v_proj.bias", "decoder.sentence_encoder.layers.4.self_attn.q_proj.weight", "decoder.sentence_encoder.layers.4.self_attn.q_proj.bias", "decoder.sentence_encoder.layers.5.self_attn.k_proj.weight", "decoder.sentence_encoder.layers.5.self_attn.k_proj.bias", "decoder.sentence_encoder.layers.5.self_attn.v_proj.weight", "decoder.sentence_encoder.layers.5.self_attn.v_proj.bias", "decoder.sentence_encoder.layers.5.self_attn.q_proj.weight", "decoder.sentence_encoder.layers.5.self_attn.q_proj.bias", "decoder.sentence_encoder.layers.6.self_attn.k_proj.weight", "decoder.sentence_encoder.layers.6.self_attn.k_proj.bias", "decoder.sentence_encoder.layers.6.self_attn.v_proj.weight", "decoder.sentence_encoder.layers.6.self_attn.v_proj.bias", "decoder.sentence_encoder.layers.6.self_attn.q_proj.weight", "decoder.sentence_encoder.layers.6.self_attn.q_proj.bias", "decoder.sentence_encoder.layers.7.self_attn.k_proj.weight", "decoder.sentence_encoder.layers.7.self_attn.k_proj.bias", "decoder.sentence_encoder.layers.7.self_attn.v_proj.weight", "decoder.sentence_encoder.layers.7.self_attn.v_proj.bias", "decoder.sentence_encoder.layers.7.self_attn.q_proj.weight", "decoder.sentence_encoder.layers.7.self_attn.q_proj.bias", "decoder.sentence_encoder.layers.8.self_attn.k_proj.weight", "decoder.sentence_encoder.layers.8.self_attn.k_proj.bias", "decoder.sentence_encoder.layers.8.self_attn.v_proj.weight", "decoder.sentence_encoder.layers.8.self_attn.v_proj.bias", "decoder.sentence_encoder.layers.8.self_attn.q_proj.weight", "decoder.sentence_encoder.layers.8.self_attn.q_proj.bias", "decoder.sentence_encoder.layers.9.self_attn.k_proj.weight", "decoder.sentence_encoder.layers.9.self_attn.k_proj.bias", "decoder.sentence_encoder.layers.9.self_attn.v_proj.weight", "decoder.sentence_encoder.layers.9.self_attn.v_proj.bias", "decoder.sentence_encoder.layers.9.self_attn.q_proj.weight", "decoder.sentence_encoder.layers.9.self_attn.q_proj.bias", "decoder.sentence_encoder.layers.10.self_attn.k_proj.weight", "decoder.sentence_encoder.layers.10.self_attn.k_proj.bias", "decoder.sentence_encoder.layers.10.self_attn.v_proj.weight", "decoder.sentence_encoder.layers.10.self_attn.v_proj.bias", "decoder.sentence_encoder.layers.10.self_attn.q_proj.weight", "decoder.sentence_encoder.layers.10.self_attn.q_proj.bias", "decoder.sentence_encoder.layers.11.self_attn.k_proj.weight", "decoder.sentence_encoder.layers.11.self_attn.k_proj.bias", "decoder.sentence_encoder.layers.11.self_attn.v_proj.weight", "decoder.sentence_encoder.layers.11.self_attn.v_proj.bias", "decoder.sentence_encoder.layers.11.self_attn.q_proj.weight", "decoder.sentence_encoder.layers.11.self_attn.q_proj.bias", "decoder.sentence_encoder.layers.12.self_attn.k_proj.weight", "decoder.sentence_encoder.layers.12.self_attn.k_proj.bias", "decoder.sentence_encoder.layers.12.self_attn.v_proj.weight", "decoder.sentence_encoder.layers.12.self_attn.v_proj.bias", "decoder.sentence_encoder.layers.12.self_attn.q_proj.weight", "decoder.sentence_encoder.layers.12.self_attn.q_proj.bias", "decoder.sentence_encoder.layers.13.self_attn.k_proj.weight", "decoder.sentence_encoder.layers.13.self_attn.k_proj.bias", "decoder.sentence_encoder.layers.13.self_attn.v_proj.weight", "decoder.sentence_encoder.layers.13.self_attn.v_proj.bias", "decoder.sentence_encoder.layers.13.self_attn.q_proj.weight", "decoder.sentence_encoder.layers.13.self_attn.q_proj.bias", "decoder.sentence_encoder.layers.14.self_attn.k_proj.weight", "decoder.sentence_encoder.layers.14.self_attn.k_proj.bias", "decoder.sentence_encoder.layers.14.self_attn.v_proj.weight", "decoder.sentence_encoder.layers.14.self_attn.v_proj.bias", "decoder.sentence_encoder.layers.14.self_attn.q_proj.weight", "decoder.sentence_encoder.layers.14.self_attn.q_proj.bias", "decoder.sentence_encoder.layers.15.self_attn.k_proj.weight", "decoder.sentence_encoder.layers.15.self_attn.k_proj.bias", "decoder.sentence_encoder.layers.15.self_attn.v_proj.weight", "decoder.sentence_encoder.layers.15.self_attn.v_proj.bias", "decoder.sentence_encoder.layers.15.self_attn.q_proj.weight", "decoder.sentence_encoder.layers.15.self_attn.q_proj.bias", "decoder.sentence_encoder.layers.16.self_attn.k_proj.weight", "decoder.sentence_encoder.layers.16.self_attn.k_proj.bias", "decoder.sentence_encoder.layers.16.self_attn.v_proj.weight", "decoder.sentence_encoder.layers.16.self_attn.v_proj.bias", "decoder.sentence_encoder.layers.16.self_attn.q_proj.weight", "decoder.sentence_encoder.layers.16.self_attn.q_proj.bias", "decoder.sentence_encoder.layers.17.self_attn.k_proj.weight", "decoder.sentence_encoder.layers.17.self_attn.k_proj.bias", "decoder.sentence_encoder.layers.17.self_attn.v_proj.weight", "decoder.sentence_encoder.layers.17.self_attn.v_proj.bias", "decoder.sentence_encoder.layers.17.self_attn.q_proj.weight", "decoder.sentence_encoder.layers.17.self_attn.q_proj.bias", "decoder.sentence_encoder.layers.18.self_attn.k_proj.weight", "decoder.sentence_encoder.layers.18.self_attn.k_proj.bias", "decoder.sentence_encoder.layers.18.self_attn.v_proj.weight", "decoder.sentence_encoder.layers.18.self_attn.v_proj.bias", "decoder.sentence_encoder.layers.18.self_attn.q_proj.weight", "decoder.sentence_encoder.layers.18.self_attn.q_proj.bias", "decoder.sentence_encoder.layers.19.self_attn.k_proj.weight", "decoder.sentence_encoder.layers.19.self_attn.k_proj.bias", "decoder.sentence_encoder.layers.19.self_attn.v_proj.weight", "decoder.sentence_encoder.layers.19.self_attn.v_proj.bias", "decoder.sentence_encoder.layers.19.self_attn.q_proj.weight", "decoder.sentence_encoder.layers.19.self_attn.q_proj.bias", "decoder.sentence_encoder.layers.20.self_attn.k_proj.weight", "decoder.sentence_encoder.layers.20.self_attn.k_proj.bias", "decoder.sentence_encoder.layers.20.self_attn.v_proj.weight", "decoder.sentence_encoder.layers.20.self_attn.v_proj.bias", "decoder.sentence_encoder.layers.20.self_attn.q_proj.weight", "decoder.sentence_encoder.layers.20.self_attn.q_proj.bias", "decoder.sentence_encoder.layers.21.self_attn.k_proj.weight", "decoder.sentence_encoder.layers.21.self_attn.k_proj.bias", "decoder.sentence_encoder.layers.21.self_attn.v_proj.weight", "decoder.sentence_encoder.layers.21.self_attn.v_proj.bias", "decoder.sentence_encoder.layers.21.self_attn.q_proj.weight", "decoder.sentence_encoder.layers.21.self_attn.q_proj.bias", "decoder.sentence_encoder.layers.22.self_attn.k_proj.weight", "decoder.sentence_encoder.layers.22.self_attn.k_proj.bias", "decoder.sentence_encoder.layers.22.self_attn.v_proj.weight", "decoder.sentence_encoder.layers.22.self_attn.v_proj.bias", "decoder.sentence_encoder.layers.22.self_attn.q_proj.weight", "decoder.sentence_encoder.layers.22.self_attn.q_proj.bias", "decoder.sentence_encoder.layers.23.self_attn.k_proj.weight", "decoder.sentence_encoder.layers.23.self_attn.k_proj.bias", "decoder.sentence_encoder.layers.23.self_attn.v_proj.weight", "decoder.sentence_encoder.layers.23.self_attn.v_proj.bias", "decoder.sentence_encoder.layers.23.self_attn.q_proj.weight", "decoder.sentence_encoder.layers.23.self_attn.q_proj.bias". 
	Unexpected key(s) in state_dict: "decoder.sentence_encoder.layers.0.self_attn.in_proj_weight", "decoder.sentence_encoder.layers.0.self_attn.in_proj_bias", "decoder.sentence_encoder.layers.1.self_attn.in_proj_weight", "decoder.sentence_encoder.layers.1.self_attn.in_proj_bias", "decoder.sentence_encoder.layers.2.self_attn.in_proj_weight", "decoder.sentence_encoder.layers.2.self_attn.in_proj_bias", "decoder.sentence_encoder.layers.3.self_attn.in_proj_weight", "decoder.sentence_encoder.layers.3.self_attn.in_proj_bias", "decoder.sentence_encoder.layers.4.self_attn.in_proj_weight", "decoder.sentence_encoder.layers.4.self_attn.in_proj_bias", "decoder.sentence_encoder.layers.5.self_attn.in_proj_weight", "decoder.sentence_encoder.layers.5.self_attn.in_proj_bias", "decoder.sentence_encoder.layers.6.self_attn.in_proj_weight", "decoder.sentence_encoder.layers.6.self_attn.in_proj_bias", "decoder.sentence_encoder.layers.7.self_attn.in_proj_weight", "decoder.sentence_encoder.layers.7.self_attn.in_proj_bias", "decoder.sentence_encoder.layers.8.self_attn.in_proj_weight", "decoder.sentence_encoder.layers.8.self_attn.in_proj_bias", "decoder.sentence_encoder.layers.9.self_attn.in_proj_weight", "decoder.sentence_encoder.layers.9.self_attn.in_proj_bias", "decoder.sentence_encoder.layers.10.self_attn.in_proj_weight", "decoder.sentence_encoder.layers.10.self_attn.in_proj_bias", "decoder.sentence_encoder.layers.11.self_attn.in_proj_weight", "decoder.sentence_encoder.layers.11.self_attn.in_proj_bias", "decoder.sentence_encoder.layers.12.self_attn.in_proj_weight", "decoder.sentence_encoder.layers.12.self_attn.in_proj_bias", "decoder.sentence_encoder.layers.13.self_attn.in_proj_weight", "decoder.sentence_encoder.layers.13.self_attn.in_proj_bias", "decoder.sentence_encoder.layers.14.self_attn.in_proj_weight", "decoder.sentence_encoder.layers.14.self_attn.in_proj_bias", "decoder.sentence_encoder.layers.15.self_attn.in_proj_weight", "decoder.sentence_encoder.layers.15.self_attn.in_proj_bias", "decoder.sentence_encoder.layers.16.self_attn.in_proj_weight", "decoder.sentence_encoder.layers.16.self_attn.in_proj_bias", "decoder.sentence_encoder.layers.17.self_attn.in_proj_weight", "decoder.sentence_encoder.layers.17.self_attn.in_proj_bias", "decoder.sentence_encoder.layers.18.self_attn.in_proj_weight", "decoder.sentence_encoder.layers.18.self_attn.in_proj_bias", "decoder.sentence_encoder.layers.19.self_attn.in_proj_weight", "decoder.sentence_encoder.layers.19.self_attn.in_proj_bias", "decoder.sentence_encoder.layers.20.self_attn.in_proj_weight", "decoder.sentence_encoder.layers.20.self_attn.in_proj_bias", "decoder.sentence_encoder.layers.21.self_attn.in_proj_weight", "decoder.sentence_encoder.layers.21.self_attn.in_proj_bias", "decoder.sentence_encoder.layers.22.self_attn.in_proj_weight", "decoder.sentence_encoder.layers.22.self_attn.in_proj_bias", "decoder.sentence_encoder.layers.23.self_attn.in_proj_weight", "decoder.sentence_encoder.layers.23.self_attn.in_proj_bias". 
```

The problem is resolved when I manually install fairseq from the commit before `fdf4c3e90`.

**Comment by akinh** [comment]
It's addressed with https://github.com/fairinternal/fairseq-py/pull/895

**Comment by myleott** [comment]
Fixed by 71b139765e66f73577fb03356e957cdc78e6f079
